Some very ugly ports away from deprecated Qt methods

nyalldawson opened the pull request:
There's no straightforward way to avoid the deprecated methods here, so we have to
copy Qt5's compatibility helpers. Ugly, but unavoidable.

3nids commented:
wouldn't it be nicer to paste them in qgis.h or maybe some other new header?
If this kind of code is centralized, it is more likely to be reused and globally replaced in the future.
But is there a requirement that it stays in the cpp files?

nyalldawson commented:
> wouldn't it be nicer to paste them in qgis.h or maybe some other new header?

We don't want to use these anywhere. Not even where they are used now :) I'm hoping someone else can find an alternative approach which avoids the pre-encoded requirements, but failing that I'd much rather hide these ugly details away from anyone using them elsewhere.

nyalldawson committed:
Some very ugly ports away from deprecated Qt methods

There's no straightforward way to avoid the deprecated methods here, so we have to
copy Qt5's compatibility helpers. Ugly, but unavoidable.
